pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -1157,7 +1165,7 @@ ASTReadPastNextOccurrence ReadPastNextOccurrence(String target) :
 ASTSqlStatement SqlStatement(String initiator, String terminator)  :
 {}
 {
-	(<SELECT>|<UPDATE>|<INSERT>|<DELETE>|<COMMIT>|<ROLLBACK>|<SAVEPOINT>|<SET><TRANSACTION>|<LOCK><TABLE>|<MERGE>|<WITH>)
+    (<SELECT>|<UPDATE>|<INSERT>|<DELETE>|<COMMIT>{jjtThis.setUserData("COMMIT"); }|<ROLLBACK>{jjtThis.setUserData("ROLLBACK"); }|<SAVEPOINT>|<SET><TRANSACTION>|<LOCK><TABLE>|<MERGE>|<WITH>)
```

Comment:
We should create the class `ASTSqlStatement` and enhance this to set the statement there ... this makes it more accessible and useful to be used - setUserData is solely based on convention. In order to use it, you need to know that ASTSqlStatement stores a string there - but other nodes can store other data...

I'd suggest to create `setType(String)` and `getType()` method there.

Please also note, that this production doesn't actually parse SQL, it just **skips** the stuff completely...